messaging/handler.go

```diff
@@ -100,7 +101,21 @@ func (h *Handler) getMessages(w http.ResponseWriter, req *http.Request) {
 
 	// Write out all data from the topic reader.
 	// Automatically flush as reads come in.
-	if _, err := CopyFlush(w, r); err != nil {
+	if _, err := CopyFlush(w, r); err == ErrTopicTruncated {
+		// Broker unable to provide data for the requested index, provide another URL.
+		urls := h.Broker.DataURLsForTopic(topicID, index)
+		if urls == nil {
+			h.error(w, ErrTopicNodesNotFound, http.StatusInternalServerError)
+		}
+
+		// Send back a list of URLs where the topic data can be fetched.
+		var redirects []string
+		for _, u := range urls {
+			redirects = append(redirects, u.String())
+		}
+		w.Header().Set("X-Broker-Truncated", strings.Join(redirects, ","))
+		http.Redirect(w, req, urls[0].String(), http.StatusTemporaryRedirect)
```

Comment:
`X-Broker-Truncated` sounds like a boolean. Can you change it to `X-Broker-DataURLs`? The truncation should be implied by the status code. Or you can set a `X-Broker-Truncated` to `true` in addition to the `DataURLs` header.